src/applications/disability-benefits/all-claims/utils.jsx

```diff
@@ -263,108 +268,212 @@ export const bankFieldsHaveInput = formData =>
-  const addressUIConfig = omit(addressUI(' '), addressOmitions);
-
-  const locationSchema = {
-    addressUI: {
-      ...addressUIConfig,
-      'ui:order': order,
-    },
-    addressSchema: {
-      ...addressSchemaConfig,
-      properties: {
-        ...omit(addressSchemaConfig.properties, addressOmitions),
-      },
-    },
-  };
-
-  if (!addressOmitions.includes('country')) {
-    locationSchema.addressUI.country = {
-      'ui:title': fieldLabels.country,
-    };
+export const AddressViewField = ({ formData }) => {
+  const {
+    addressLine1,
+    addressLine2,
+    addressLine3,
+    city,
+    country,
+    state,
+    zipCode,
+  } = formData;
+  let zipString;
+  if (zipCode) {
+    const firstFive = zipCode.slice(0, 5);
+    const lastChunk = zipCode.length > 5 ? `-${zipCode.slice(5)}` : '';
+    zipString = `${firstFive}${lastChunk}`;
   }
 
-  if (!addressOmitions.includes('addressLine1')) {
-    locationSchema.addressUI.addressLine1 = {
-      'ui:title': fieldLabels.addressLine1,
-    };
+  let lastLine;
+  if (country === USA) {
+    lastLine = `${city}, ${state} ${zipString}`;
+  } else {
+    lastLine = `${city}, ${country}`;
   }
+  return (
+    <p className="blue-bar-block">
+      {addressLine1 && addressLine1}
+      <br />
+      {addressLine2 && addressLine2}
+      {addressLine2 && <br />}
+      {addressLine3 && addressLine3}
+      {addressLine3 && <br />}
+      {lastLine}
+    </p>
+  );
+};
 
-  if (!addressOmitions.includes('addressLine2')) {
-    locationSchema.addressUI.addressLine2 = {
-      'ui:title': fieldLabels.addressLine2,
-    };
-  }
+/**
+ * Returns the path with any ':index' substituted with the actual index.
+ * @param {string} path - The path with or without ':index'
+ * @param {number} index - The index to put in the string
+ * @return {string}
+ */
+export const pathWithIndex = (path, index) => path.replace(':index', index);
 
-  if (!addressOmitions.includes('city')) {
-    locationSchema.addressUI.city = {
-      'ui:title': fieldLabels.city,
-    };
-  }
+/**
+ * Returns the uiSchema for addresses that use the non-common address schema as found
+ *  in the 526EZ-all-claims schema.
+ * @param {string} addressPath - The path to the address in the formData
+ * @param {string} [title] - Displayed as the card title in the card's header
+ * @param {boolean} reviewCard - Whether to display the information in a ReviewCardField or not
+ * @param {boolean} fieldsAreRequired - Whether the typical fields should be required or not
+ * @returns {object} - UI schema for an address card's content
+ */
+export const addressUISchema = (
+  addressPath,
+  title,
+  reviewCard,
+  fieldsAreRequired = true,
+) => {
+  const updateStates = (formData, currentSchema, uiSchema, index) => {
+    // Could use path (updateSchema callback param after index), but it points to `state`,
+    //  so using `addressPath` is easier
+    const currentCity = _.get(
+      `${pathWithIndex(addressPath, index)}.city`,
```

Comment:
The other main change to this function is the addition of `pathWithIndex` so we can use it in the 8940 without trouble since the addresses there are in arrays.